tools/testutils.py

```diff
@@ -17,133 +17,160 @@
+    log.info(f"Writing {args}")
     proc = None
 
     try:
-        proc = Popen(args,
-                     stdout=subprocess.PIPE,
-                     shell=True,
-                     stdin=subprocess.PIPE,
-                     stderr=subprocess.PIPE,
-                     universal_newlines=True)
+        proc = subprocess.Popen(
+            args,
+            stdout=subprocess.PIPE,
+            shell=True,
+            stdin=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,
+            preexec_fn=os.setsid,
+        )
     except OSError as e:
-        report_err(outputs["stderr"], "Failed executing: ", e)
+        log.error(f"Failed executing:\n{e}")
     if proc is None:
         # Never even started
-        report_err(outputs["stderr"], "Process failed to start")
+        log.error("Process failed to start")
     return proc
 
 
-def run_process(verbose, proc, timeout, outputs, errmsg):
-
+def run_process(proc, timeout, errmsg):
     def kill(process):
         process.kill()
 
-    timer = Timer(TIMEOUT, kill, [proc])
+    timer = Timer(timeout, kill, [proc])
     try:
         timer.start()
         out, err = proc.communicate()
     finally:
         timer.cancel()
     if out:
-        msg = ("\n########### PROCESS OUTPUT BEGIN:\n"
-               "%s########### PROCESS OUTPUT END\n" % out)
-        report_output(outputs["stdout"], verbose, msg)
+        msg = "\n########### PROCESS OUTPUT BEGIN:\n" f"{out}########### PROCESS OUTPUT END\n"
+        log.info(msg)
     if proc.returncode != SUCCESS:
-        report_err(outputs["stderr"],
-                   "Error %d: %s\n%s" % (proc.returncode, errmsg, err))
+        log.error(f"Error {proc.returncode}: {errmsg}\n{err}")
     else:
         # Also report non fatal warnings in stdout
         if err:
-            report_err(outputs["stderr"], err)
+            log.error(err)
     return proc.returncode
 
 
-def run_timeout(verbose, args, timeout, outputs, errmsg):
-    proc = open_process(verbose, args, outputs)
-    if proc is None:
-        return FAILURE
-    report_output(outputs["stdout"], verbose, "Executing", args)
-    return run_process(verbose, proc, timeout, outputs, errmsg)
+def exec_process(args, errmsg, timeout=TIMEOUT):
+    log.debug(f"Executing command: {args}")
+    output_args = {}
+    if log.getEffectiveLevel() <= logging.INFO:
+        output_args = {"stdout": sys.stdout, "stderr": sys.stdout}
+    else:
+        output_args = {"stdout": subprocess.PIPE, "stderr": sys.stdout}
+    try:
+        result = subprocess.run(
+            args, shell=True, timeout=timeout, universal_newlines=True, check=True, **output_args
+        )
+    except subprocess.CalledProcessError as e:
+        out = e.stdout
+        returncode = e.returncode
+        log.error(f"Error {returncode} when executing {e.cmd}:\n{errmsg}\n{out}")
+    else:
+        out = result.stdout
+        returncode = result.returncode
```

Comment:
I'd suggest putting this into the `try` block, it is more readable and natural and this lines of code cannot raise `CalledProcessError `.